framework/PostProcessors/ETimporter.py

```diff
@@ -0,0 +1,554 @@
+        The variable self.links=[dep1,...,depN] is a list of connections where each connection dep is a
+        dictionary as follows:
+
+            dep.keys =
+                       * link_seqID  : ID of the link in the master ET
+                       * ET_slave_ID : ID of the slave ET that needs to be copied into the master ET
+                       * ET_master_ID: ID of the master ET;
+
+        The slave ET is merged into the master ET; note the master ET contains the link in at least one
+        <define-sequence> node:
+
+           <define-sequence name="Link-to-LP">
+             <event-tree name="Link-to-LP-Event-Tree"/>
+           </define-sequence>
+
+        @ In,  None
+        @ Out, None
+      """
+      self.links = []   # here I am basically deleting what was previously contained in self.links
+      for root in self.listRoots:
+          links, seqID = self.checkLinkedTree(root)
+          if len(links) > 0:
+              for idx, val in enumerate(links):
+                  dep = {}
+                  dep['link_seqID'] = copy.deepcopy(seqID[idx])
+                  dep['ET_slave_ID'] = copy.deepcopy(val)
+                  dep['ET_master_ID'] = copy.deepcopy(root.get('name'))
+                  self.links.append(dep)
+
+
+  def checkETstructure(self):
+      """
+        This method checks that the structure of the ET is consistent. In particular, it checks that only one root ET
+        and at least one leaf ET is provided. As an example consider the following ET structure:
+           ET1 ----> ET2 ----> ET3
+            |------> ET4 ----> ET5
+        Five ETs have been provided, ET1 is the only root ET while ET3 and ET5 are leaf ET.
+        @ In,  None
+        @ Out, None
+      """
+
+      # each element (i,j) of the matrix self.connectivityMatrix shows if there is a connection from ET_i to ET_j:
+      #   * 0: no connection from i to j
+      #   * 1: a connection exists from i to j
+      for link in self.links:
+          row = self.listETs.index(link['ET_master_ID'])
+          col = self.listETs.index(link['ET_slave_ID'])
+          self.connectivityMatrix[row,col]=1.0
+
+      # the root ETs are charaterized by a column full of zeros
+      # the leaf ETs are charaterized by a row full of zeros
+      zeroRows    = np.where(~self.connectivityMatrix.any(axis=1))[0]
+      zeroColumns = np.where(~self.connectivityMatrix.any(axis=0))[0]
+
+      if len(zeroColumns)>1:
+          self.raiseAnError(IOError, 'Multiple root ET')
+      if len(zeroColumns)==0:
+          self.raiseAnError(IOError, 'No root ET')
+      if len(zeroColumns)==1:
+          self.rootET_ID = self.listETs[zeroColumns]
+          self.raiseADebug("ETImporter Root ET: " + str(self.rootET_ID))
+
+      leafs = []
+      for index in np.nditer(zeroRows):
+          leafs.append(self.listETs[index])
+      self.raiseADebug("ETImporter leaf ETs: " + str(leafs))
+
+
+  def analyzeMultipleET(self,input):
```

Comment:
I suggest to use a different name for input, because input is a function in python

Reply:
state variable should be avoided in this method, such as self.listET, ..., please use local variables